### PR TITLE
ST-384 - Skeleton for CY+1

### PR DESCRIPTION
**Pull request by SeanArmstrong**
- [x] Unit tests passing
- [x] Coverage reached
- [x] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
+import uk.gov.hmrc.tai.model.TaxYear
+import uk.gov.hmrc.tai.util.constants.journeyCache.UpdateNextYearsIncomeConstants
+import uk.gov.hmrc.tai.model.cache.UpdateNextYearsIncomeCacheModel
+import scala.concurrent.Future
+
+class UpdateNextYearsIncomeService {
```
Comment (lewey) [inline app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala]: This class seems to have two responsibilities. Acting as a service layer and a cache.

File: app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  lazy val journeyCacheService: JourneyCacheService = JourneyCacheService(UpdateNextYearsIncomeConstants.JOURNEY_KEY)
+  lazy val employmentService: EmploymentService = EmploymentService
+  lazy val taxAccountService: TaxAccountService = TaxAccountService
+
+  def setup(employmentId: Int, nino: Nino)(implicit hc: HeaderCarrier): Future[UpdateNextYearsIncomeCacheModel] = {
```
Comment (lewey) [inline app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala]: There should ideally be private or nested within the get method

File: app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      }
+      case _ => throw new RuntimeException("[UpdateNextYearsIncomeService] Could not set up next years estimated income journey")
+    }
+  }
+
+  def get(employmentId: Int, nino: Nino)(implicit hc: HeaderCarrier): Future[UpdateNextYearsIncomeCacheModel] = {
+    journeyCacheService.currentCache flatMap {
+      case cache: Map[String, String] if cache.isEmpty => setup(employmentId, nino)
+      case cache: Map[String, String] => {
+        if (cache.contains(UpdateNextYearsIncomeConstants.NEW_AMOUNT)) {
```
Comment (lewey) [inline app/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeService.scala]: This code could potentially be improved if a solution is thought of for the multiple employments issue.

**Comment by amaalali** [comment]
@lewey Good points, I think we'll address in the PR for `edit` where the implementation will be more obvious

**Pull request merged**